src/Silex/Provider/VarDumperServiceProvider.php

```diff
@@ -45,13 +50,14 @@ public function boot(Application $app)
             return;
         }
 
-        // This code is here to lazy load the dump stack. This default
-        // configuration for CLI mode is overridden in HTTP mode on
-        // 'kernel.request' event
+        // This code is here to lazy load the dump stack.
         VarDumper::setHandler(function ($var) use ($app) {
-            VarDumper::setHandler($handler = function ($var) use ($app) {
-                $app['var_dumper.cli_dumper']->dump($app['var_dumper.cloner']->cloneVar($var));
-            });
+            $handler = function ($var) use ($app) {
+                'cli' === PHP_SAPI
```

Comment:
while optimal, this makes it also near impossible to utest as the value of PHP_SAPI cannot be changed @ runtime.